// crud/index.js
const mysql = require('mysql');

/*class ClassCrud{

    constructor()
    {
       this.sql = mysql.createPool({
            user: "b1e9e740318ae2",
            password: "8aeb001a",
            host: "us-cdbr-east-05.cleardb.net",
            database: "heroku_860cbb4844f3f93"
        });
    }

    read(req, res)
    {
        if (!req.params.id) {
            this.sql.getConnection((err, connection) => {
                connection.query("select * from user order by id asc", (err, results, fields) => {
                    res.render('select', {
                        data: results,
                        style: 'table.css'
                    });
                });
            });
        }
        else {
            this.sql.getConnection((err, connection) => {
                connection.query("select * from user where id=? order by id asc", [req.params.id], (err, results, fields) => {
                    res.render('select', {
                        data: results,
                        style: 'table.css'
                    });
                });
            });
        }
    }

    create(req, res)
    {
        this.sql.getConnection((err, connection) => {
            connection.query("insert into user values(?, ?, ?)", [
                req.body.id,
                req.body.name,
                req.body.age
            ]);
            res.render("controllerForm", {
                name: req.body.name,
                style: 'success.css'
            });
        });  
    }

    deletes(req, res)
    {
        this.sql.getConnection((err, connection) => {
            connection.query("delete from user where id=?", [req.params.id]);
            res.render('deletar', {
                style: 'success.css'
            });
        });
    }

    update(req, res, controller=null)
    {
        if(controller==null) {
            this.sql.getConnection((err, connection) => {
                connection.query("select * from user where id=?", [req.params.id], 
                    (err, results, fields) => {
                        res.render('update', { 
                        id: req.params.id, 
                        name: results[0].name, 
                        age: results[0].age,
                        style: 'form.css' });
                    });
            });
        }else {
            this.sql.getConnection((err, connection) => {
                connection.query("update user set name=?,age=? where id=?", 
                    [req.body.name, 
                        req.body.age, 
                        req.body.id]);
                    res.render('controllerUpdate', {
                        style: 'success.css'
                    });
            });
        }
    }
}

module.exports=ClassCrud;*/
